data/downloader/downloader.go

```diff
@@ -131,20 +131,21 @@ func (d *Downloader) handleImport(item *DownloadItem) {
 			d.failedQueue[item.URI] = item
 			d.failedQueueLock.Unlock()
 		} else {
-			log.Warnw("could not pin file", "uri", item.URI, "error", fmt.Sprintf("%v", err))
+			log.Warnw("could not retrieve file", "uri", item.URI, "error", fmt.Sprintf("%v", err))
 		}
 		return
 	}
-	ctx, cancel = context.WithTimeout(context.Background(), ImportPinTimeout)
-	defer cancel()
-	data, err := d.RemoteStorage.Retrieve(ctx, item.URI, 0)
-	if err != nil {
-		log.Warnw("could not retrieve file", "uri", item.URI, "error", fmt.Sprintf("%v", err))
-	} else {
-		if item.Callback != nil {
-			go item.Callback(item.URI, data)
+	go func() {
```

Comment:
why a goroutine now? we should start documenting these; otherwise we're sprinkling goroutines everywhere and it's hard to keep track why they're needed.

Reply:
The main objective of this callback is to retrieve the data, this is done Sync. 
Pining is a "nice-to-have" and not required for the own node (but useful for others so they can fetch also the data from other sources). This is the reasoning behind the pin.

I'll document it.